bgpd/bgp_evpn.c

```diff
@@ -2480,8 +2508,27 @@ static int install_evpn_route_entry_in_vrf(struct bgp *bgp_vrf,
 		pi->attr = attr_new;
 		pi->uptime = bgp_clock();
 	}
-	/* as it is an importation, change nexthop */
-	bgp_path_info_set_flag(dest, pi, BGP_PATH_ANNC_NH_SELF);
+
+	/* Gateway IP nexthop should be resolved */
+	if (attr.evpn_overlay.type == OVERLAY_INDEX_GATEWAY_IP) {
+		if (bgp_find_or_add_nexthop(bgp_vrf, bgp_vrf, afi, safi, pi,
```

Comment:
overlay gw-ip seems like an ideal candidate for the NHG indirection, right? Did you consider that design?